src/drivers/dw/dma.c

```diff
@@ -1290,6 +1279,18 @@ static int dw_dma_probe(struct dma *dma)
 		return -ENOMEM;
 	}
 	dma_set_drvdata(dma, dw_pdata);
+#endif
+
+	/* allocate dma channels */
+	dma->chan = rzalloc(RZONE_SYS_RUNTIME | RZONE_FLAG_UNCACHED,
+			    SOF_MEM_CAPS_RAM,
+			    sizeof(struct dma_chan_data) * DW_MAX_CHAN);
```

Comment:
We don't need to allocate for `DW_MAX_CHAN` anymore. Now we can read actual number of channels from `dma->plat_data.channels`.

Reply:
@tlauda done